### ISSUE TITLE
how to use sys.proc.spawn?

### ISSUE THREAD

**Issue by nivpgir** [TurtleKitty/Vaquero#1]
I want to run a process and get back it's stdin stdout and stderr, as far as I can see sys.proc.spawn is supposed to give it to me but I can't seem to figure out how to work with it.
some failed attempts:
```
(def echo (sys.proc.spawn "echo stuff and more stuff"))
(say echo.pid) ; prints the pid
(say echo.stdout.read-line) ; prints EOF
```
after more trial and error I realized the command isn't really being run, so I tried running it:
```
(def echo (sys.proc.spawn "echo stuff and more stuff") (lambda (o) (say o.pid))) ; maybe it gets a callback
```
```
(def echo (sys.proc.spawn "echo stuff and more stuff") #f (lambda (o) (say o.pid)) #f) ; maybe it gets a callback in the `cont` position as in the source code
```

```
(def echo (sys.proc.spawn "echo stuff and more stuff")) 
(apply  echo ())
(echo.proc-apply)
(echo)
```
... and more similarly crazy attempts

**Comment by TurtleKitty** [comment]
You know... now that you mention it, I don't think I've ever gotten that to work right.  Looks like I left it out of the docs on the wiki.  `sys.proc.fork` works, but requires one to set up your own sockets and such (some examples can be found in `tests/other/fork.vaq` and `tests/other/sockets.vaq`).

**Comment by nivpgir** [comment]
I've made some progress with this.
basically it's because if chickens `process*` gets more than 1 argument, then the first and second arguments need to be a command and a list of strings which will be it's arguments. and since thats the only way `process*` is being called (always with an empty list as args for the command, the sys.proc.spawn function can only work with a single binary as command and no arguments.

I've managed to get it to work like this:
```
(let ((cmd (car args))
		     (cmd-args  (cdr args)))
		 (define env (if (hte? opts 'env) (hash-table->alist (htr opts 'env)) '()))
                  (define-values (stdout stdin pid stderr) (process* cmd cmd-args env))
                   ...
```

but that requires that the input to `spawn` will be a list of strings instead of a single string (splitting the string isn't good enough because sometimes an argument will contain spaces in it)).
that's the best I've got so far.

Another annoying thing which I noticed is that the command has to be a full path to the executable, i.e `echo` doesn't work but `/bin/echo` does, I don't know how to fix that other than resolve commands to full path inside spawn.
